esphome/components/ltr_als_ps/ltr_als_ps.cpp

```diff
@@ -0,0 +1,520 @@
+#include "ltr_als_ps.h"
+#include "esphome/core/application.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+using esphome::i2c::ErrorCode;
+
+namespace esphome {
+namespace ltr_als_ps {
+
+static const char *const TAG = "ltr_als_ps";
+
+static const uint8_t MAX_TRIES = 5;
+
+template<typename T, size_t size> T get_next(const T (&array)[size], const T val) {
+  size_t i = 0;
+  size_t idx = -1;
+  while (idx == -1 && i < size) {
+    if (array[i] == val) {
+      idx = i;
+      break;
+    }
+    i++;
+  }
+  if (idx == -1 || i + 1 >= size)
+    return val;
+  return array[i + 1];
+}
+
+template<typename T, size_t size> T get_prev(const T (&array)[size], const T val) {
+  size_t i = size - 1;
+  size_t idx = -1;
+  while (idx == -1 && i > 0) {
+    if (array[i] == val) {
+      idx = i;
+      break;
+    }
+    i--;
+  }
+  if (idx == -1 || i == 0)
+    return val;
+  return array[i - 1];
+}
+
+static uint16_t get_itime_ms(IntegrationTime time) {
+  static const uint16_t ALS_INT_TIME[8] = {100, 50, 200, 400, 150, 250, 300, 350};
+  return ALS_INT_TIME[time & 0b111];
+}
+
+static uint16_t get_meas_time_ms(MeasurementRepeatRate rate) {
+  static const uint16_t ALS_MEAS_RATE[8] = {50, 100, 200, 500, 1000, 2000, 2000, 2000};
+  return ALS_MEAS_RATE[rate & 0b111];
+}
+
+static float get_gain_coeff(AlsGain gain) {
+  static const float ALS_GAIN[8] = {1, 2, 4, 8, 0, 0, 48, 96};
+  return ALS_GAIN[gain & 0b111];
+}
+
+static float get_ps_gain_coeff(PsGain gain) {
+  static const float PS_GAIN[4] = {16, 0, 32, 64};
+  return PS_GAIN[gain & 0b11];
+}
+
+void LTRAlsPsComponent::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up LTR-303/329");
```

Comment:
I wonder if this should be expanded somehow to included all supported sensors; maybe something like this?:
```suggestion
  ESP_LOGCONFIG(TAG, "Setting up LTR-303/329/55x/659");
```